packages/bruno-app/src/components/Sidebar/Collections/index.js

```diff
@@ -1,71 +1,96 @@
+import { IconSearch, IconX } from '@tabler/icons';
+import Collection from '../Collections/Collection';
 import CreateCollection from '../CreateCollection';
-import Collection from './Collection';
-import CollectionsHeader from './CollectionsHeader';
-import CreateOrOpenCollection from './CreateOrOpenCollection';
 import StyledWrapper from './StyledWrapper';
+import CreateOrOpenCollection from './CreateOrOpenCollection';
 
-const Collections = () => {
+const Collections = ({ showSearch }) => {
   const [searchText, setSearchText] = useState('');
   const { collections } = useSelector((state) => state.collections);
+  const { workspaces, activeWorkspaceUid } = useSelector((state) => state.workspaces);
   const [createCollectionModalOpen, setCreateCollectionModalOpen] = useState(false);
 
-  if (!collections || !collections.length) {
+  const activeWorkspace = workspaces.find((w) => w.uid === activeWorkspaceUid) || workspaces.find((w) => w.type === 'default');
+
+  let allCollections = [];
+
+  if (!activeWorkspace || activeWorkspace.type === 'default') {
+    if (activeWorkspace && activeWorkspace.collections && activeWorkspace.collections.length > 0) {
+      allCollections = activeWorkspace.collections.map((wc) => {
+        const loadedCollection = collections.find((c) => c.pathname === wc.path);
+        return loadedCollection;
+      }).filter(Boolean);
+    } else {
+      allCollections = [];
+    }
+  } else {
+    if (activeWorkspace.collections && activeWorkspace.collections.length > 0) {
+      allCollections = activeWorkspace.collections.map((wc) => {
+        const loadedCollection = collections.find((c) => c.pathname === wc.path);
+        return loadedCollection;
+      }).filter(Boolean);
+    }
+  }
+
+  if (!allCollections || !allCollections.length) {
     return (
-      <StyledWrapper data-testid="collections">
-        <CollectionsHeader />
+      <StyledWrapper>
         <CreateOrOpenCollection />
       </StyledWrapper>
     );
   }
 
   return (
     <StyledWrapper data-testid="collections">
-      {createCollectionModalOpen ? <CreateCollection onClose={() => setCreateCollectionModalOpen(false)} /> : null}
-
-      <CollectionsHeader />
-
-      <div className="mt-4 relative collection-filter px-2">
-        <div className="absolute inset-y-0 left-0 pl-4 flex items-center pointer-events-none">
-          <span className="text-gray-500">
-            <IconSearch size={16} strokeWidth={1.5} />
-          </span>
-        </div>
-        <input
-          type="text"
-          name="search"
-          placeholder="Search requests …"
-          id="search"
-          autoComplete="off"
-          autoCorrect="off"
-          autoCapitalize="off"
-          spellCheck="false"
-          className="block w-full pl-7 py-1"
-          value={searchText}
-          onChange={(e) => setSearchText(e.target.value.toLowerCase())}
+      {createCollectionModalOpen ? (
+        <CreateCollection
```

Comment:
The previous code was not passing anything else to `CreateCollection` except `onClose`

We should not be passing down `workspaceUid` and `hideLocationInput` as this can be resolved by the CreateCollection itself from the global redux store. 

Two advantages
1. We avoid prop drilling
2. `CreateCollection` can be used in mulltiple places, and we don't need to go do prop drilling at multiple places in the future